inference-engine/src/mkldnn_plugin/nodes/mkldnn_if_node.cpp

```diff
@@ -170,57 +167,85 @@ void MKLDNNIfNode::initSupportedPrimitiveDescriptors() {
     supportedPrimitiveDescriptors.emplace_back(config, impl_desc_type::unknown);
 }
 
-
 void MKLDNNIfNode::createPrimitive() {
-    const auto& eng = getEngine();
-
-    for (auto& map_rule : thenInputPortMap) {
-        auto &fromMem = getParentEdgesAtPort(map_rule.from)[0]->getMemoryPtr();
-        auto &toMem = inputMemThen[map_rule.to];
-
-        beforeThenMappers.emplace_back(std::make_shared<PortMapHelper>(fromMem, toMem, eng));
+    if (inputShapesDefined()) {
+        if (needPrepareParams())
+            prepareParams();
+        updateLastInputDims();
     }
+}
 
-    for (auto& map_rule : elseInputPortMap) {
-        auto &fromMem = getParentEdgesAtPort(map_rule.from)[0]->getMemoryPtr();
-        auto &toMem = inputMemElse[map_rule.to];
+bool MKLDNNIfNode::needPrepareParams() const {
+    return MKLDNNNode::needPrepareParams() || condition != (reinterpret_cast<const bool*>(getParentEdgeAt(0)->getMemoryPtr()->GetPtr()))[0];
+}
 
-        beforeElseMappers.emplace_back(std::make_shared<PortMapHelper>(fromMem, toMem, eng));
+void MKLDNNIfNode::prepareParams() {
+    const auto& eng = getEngine();
+
+    if (isDynamic) {
```

Comment:
The fields of `PortMapHelper` are `from`, `to` memories and reorder which are initialized inside ctor. After each infer request with dynamism support input shapes,  `from`  and `to` memories (shared pointers) are changed. So all fields of this class may be changed. I think we can create these helpers after each inference request because shared pointer creation isn't expensive